### PR TITLE
plan: refine UnionAll building

### PR DESCRIPTION
**Pull request by hanfei1991**
I uniform the schema of ua and its children to avoid extra cost.
and physical union all cannot pass order prop.

### CODE REVIEW

File: plan/logical_plan_builder.go [hunk 1]
```diff
@@ -587,12 +587,28 @@ func joinFieldType(a, b *types.FieldType) *types.FieldType {
 }
 
 func (b *planBuilder) buildProjection4Union(u *LogicalUnionAll) {
-	schema4Union := u.schema
+	unionSchema := u.children[0].Schema().Clone()
+
+	// infer union type
```
Comment (winoros) [inline plan/logical_plan_builder.go]: Make comment more formal.

**Comment by winoros** [comment]
rest LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**